packages/multi-select-combo-box/src/vaadin-multi-select-combo-box.js

```diff
@@ -834,10 +843,27 @@ class MultiSelectComboBox extends ResizeMixin(InputControlMixin(ThemableMixin(El
     return selectedItems.indexOf(item);
   }
 
-  /** @private */
-  __clearFilter() {
-    this.filter = '';
-    this.$.comboBox.clear();
+  /**
+   * Clear the internal combo box value and filter. Filter will not be cleared
+   * when the `keepFilter` option is enabled. Using `force` can enforce clearing
+   * the filter.
+   * @param {boolean} force overrides the keepFilter option
+   * @private
+   */
+  __clearInternalValue(force = false) {
+    if (!this.keepFilter || force) {
+      // Clear both combo box value and filter.
+      this.filter = '';
+      this.$.comboBox.clear();
+    } else {
+      // Only clear combo box value. This effectively resets _lastCommittedValue
+      // which allows toggling the same item multiple times via keyboard.
+      this.$.comboBox.clear();
+      // Restore input to the filter value. Needed when items are
+      // navigated with keyboard, which overrides the input value
+      // with the item label.
+      this.inputElement.value = this.filter;
```

Comment:
Replaced the single occurence of `this._inputElementValue` with `this.inputElement.value`.